pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -4,17 +4,92 @@ package dynamichelper
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+	"errors"
 	"reflect"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
 )
 
+func TestEsureDeleted(t *testing.T) {
```

Comment:
We need a test case to check that [not found](https://github.com/Azure/ARO-RP/blob/d673482fb835b02e4b00dfe0e06135be143c3d07/pkg/util/dynamichelper/dynamichelper.go#L75-L77) condition is handled properly.